src/pose_estimation/capture_device.py

```diff
@@ -0,0 +1,38 @@
+import cv2
+from datetime import datetime
+
+class CaptureDevice:
+    def __init__(self, filepath: str, live: bool = False):
+        self.device = cv2.VideoCapture(filepath)
+        self.live = live
+        self.init_time = datetime.now()
+
+    def get_timestamp(self) -> datetime:
+        '''
+        Find the timestamp in milliseconds from the start of the video
+        '''
+        return datetime.now() - self.init_time if self.live else self.device.get(cv2.CAP_PROP_POS_MSEC)
```

Comment:
CAP_PROP_POP_MSEC doesn't for livestreams as you've seen. During research I found that time.perf_counter() is probably our best bet.

`time.perf_counter()`